public/app/features/manage-dashboards/components/RestoreModal.tsx

```diff
@@ -0,0 +1,45 @@
+import React from 'react';
+
+import { ConfirmModal, Text } from '@grafana/ui';
+
+import { Trans, t } from '../../../core/internationalization';
+import { DescendantCount } from '../../browse-dashboards/components/BrowseActions/DescendantCount';
+import { DashboardTreeSelection } from '../../browse-dashboards/types';
+
+interface Props {
+  isOpen: boolean;
+  onConfirm: () => Promise<void>;
+  onDismiss: () => void;
+  selectedItems: DashboardTreeSelection;
+  isLoading: boolean;
+}
+
+export const RestoreModal = ({ onConfirm, onDismiss, selectedItems, isLoading, ...props }: Props) => {
+  const onRestore = async () => {
+    await onConfirm();
+    onDismiss();
+  };
+  return (
+    <ConfirmModal
+      body={
+        <>
+          <Text element="p">
+            <Trans i18nKey="recentlyDeleted.restoreModal.text">This action will restore the following content:</Trans>
+          </Text>
+          <DescendantCount selectedItems={selectedItems} />{' '}
+          {/*TODO: clarify whether "3 items: 3 dashboards" makes sense */}
```

Comment:
We don't need DescendantCount here because Dashboards cannot have descendants, only folders can!

You could change the test to 

```ts
const selectedCount = selectedItems.length
...
<Trans i18nKey="recentlyDeleted.restoreModal.text" count={selectedCount}>This action will restore {{selectedCount}} dashboards</Trans>
```

Reply:
That's what I was looking into and I wanted to change the text as well. So I got in contact with someone from the docs team again and we agreed on the new version.